### DISCUSSION TITLE
Option to change pinmode for IRSend ?

### DISCUSSION THREAD

**Discussion by mahipat99** [arendst/Tasmota#16434] [Q&A] [answered]
Hello,

Is it possible to introduce change pinMode for IRSend via my_config?
Like changing invert and duty cycle via `IR_SEND_INVERTED` & `IR_SEND_USE_MODULATION.`

For the test, I have made changes in lib, and it's working as expected
https://github.com/arendst/Tasmota/blob/e71235ea3a397c0cd4593c33606b6778bf8fb66d/lib/lib_basic/IRremoteESP8266/IRremoteESP8266/src/IRsend.cpp#L47
`pinMode(IRpin, OUTPUT_OPEN_DRAIN);`

**Comment by s-hadinger** [comment, marked as answer]
This code is not part of Tasmota. Please submit the request to the IRREMOTEESP8266 repository. 

Why do you need this?

**Reply by barbudor** [reply]
I Imagine he is driving the IR led directly from the ESP pin instead of going through a transistor as recommended.

**Reply by mahipat99** [reply]
> This code is not part of Tasmota

Oops, my bad! Thanks for the heads-up
Sending IR command directly to IR RX with 100% duty cycle

**Reply by sfromis** [reply]
If barbudor is right about you trying to drive a IR led directly without a transistor, beware that you may get rather bad range.

**Reply by s-hadinger** [reply]
Then why is OUTPUT not good? Does it harm if the GPIO provides 3.3V?

Just to clarify my questions. I want to know if it's a common pattern that we will see with other devices or is it a specific design for a specific case that nobody will ever need again.

**Reply by mahipat99** [reply]
Thanks for your concern @barbudor @sfromis 
But I am sending data directly to IR Receiver. You can say mimic/emulate instead of sending the signal via IR LED.

@s-hadinger 
No, but if I set it as output, the pin will always be high (carrier duty 100%), and the original remote will stop working. Only emulated signal via ESP device will work

**Reply by sfromis** [reply]
Does not sound like a "common pattern" then....

**Reply by s-hadinger** [reply]
Then I propose that you patch directly in the code and self-compile
